l10n_it_fatturapa_import_zip/models/attachment.py

```diff
@@ -0,0 +1,550 @@
+                % (company.name, company.id)
+            )
+        return journals[0]
+
+    def get_account_taxes(self, AliquotaIVA, Natura):
+        account_tax_model = self.env["account.tax"]
+        wizard_import = self.env["wizard.import.fatturapa"]
+        ir_values = self.env["ir.default"]
+        company_id = self.env.company.id
+        taxes_ids = ir_values.get("product.product", "taxes_id", company_id=company_id)
+        def_tax = False
+        if taxes_ids:
+            def_tax = account_tax_model.browse(taxes_ids, limit=1)
+        if float(AliquotaIVA) == 0.0 and Natura:
+            account_taxes = account_tax_model.search(
+                [
+                    ("type_tax_use", "=", "sale"),
+                    ("kind_id.code", "=", Natura),
+                    ("amount", "=", 0.0),
+                ],
+                order="sequence",
+            )
+            if not account_taxes:
+                wizard_import.log_inconsistency(
+                    _(
+                        "No tax with percentage "
+                        "%s and nature %s found. Please configure this tax."
+                    )
+                    % (AliquotaIVA, Natura)
+                )
+            if len(account_taxes) > 1:
+                wizard_import.log_inconsistency(
+                    _(
+                        "Too many taxes with percentage "
+                        "%s and nature %s found. Tax %s with lower priority has "
+                        "been set on invoice lines."
+                    )
+                    % (AliquotaIVA, Natura, account_taxes[0].description)
+                )
+        else:
+            account_taxes = account_tax_model.search(
+                [
+                    ("type_tax_use", "=", "sale"),
+                    ("amount", "=", float(AliquotaIVA)),
+                    ("price_include", "=", False),
+                    ("children_tax_ids", "=", False),
+                ],
+                order="sequence",
+            )
+            if not account_taxes:
+                wizard_import.log_inconsistency(
+                    _(
+                        "XML contains tax with percentage '%s' "
+                        "but it does not exist in your system"
+                    )
+                    % AliquotaIVA
+                )
+            if len(account_taxes) > 1:
+                if def_tax and def_tax.amount == (float(AliquotaIVA)):
+                    account_taxes = def_tax
+        return account_taxes
+
+    def _prepare_generic_line_data(self, line):
+        retLine = {}
+        account_taxes = self.get_account_taxes(line.AliquotaIVA, line.Natura)
+        if account_taxes:
+            retLine["tax_ids"] = [(6, 0, [account_taxes[0].id])]
+        return retLine
+
+    def _prepareInvoiceLine(self, debit_account_id, line):
+        wizard_import = self.env["wizard.import.fatturapa"]
+        retLine = self._prepare_generic_line_data(line)
```

Comment:
qui potresti chiamare il metodo con lo stesso nome in `wizard_import`.
In generale vale quanto scrivo poco sopra: viene introdotto troppo codice duplicato.

Reply:
In generale vale quanto scrivo poco sopra: dato che la `_prepare_generic_line_data` va a cercare la `get_account_taxes`